### PR TITLE
new message interface

### PR DESCRIPTION
**Pull request by huzaif-plivo**

### CODE REVIEW

File: src/Plivo/Resources/Message/MessageInterface.php [hunk 1]
```diff
@@ -76,45 +64,45 @@ public function get($messageUuid)
...
-        foreach ($response->getContent()['objects'] as $message) {
+        foreach ($response->getContent() ['objects'] as $message)
+        {
             $newMessage = new Message($this->client, $message, $this->pathParams['authId'], $this->uri);
 
             array_push($messages, $newMessage);
         }
 
-        return new MessageList($this->client, $response->getContent()['meta'], $messages);
+        return new MessageList($this->client, $response->getContent() ['meta'], $messages);
```
Comment (narayana-plivo) [inline src/Plivo/Resources/Message/MessageInterface.php:80]: do we need a space here ?$response->getContent() ['meta']

File: src/Plivo/Resources/Message/MessageInterface.php [hunk 2]
```diff
@@ -144,66 +132,101 @@ protected function getList($optionalArgs = [])
      * @throws PlivoValidationException,PlivoResponseException
      */
 
-    public function create($src, array $dst, $text=null,
-                           array $optionalArgs = [], $powerpackUUID = null)
+    public function __call($name_of_method, $arguments)
```
Comment (narayana-plivo) [inline src/Plivo/Resources/Message/MessageInterface.php:135]: why there is function signature change?  @nixonsam  Please check once

Comment (narayana-plivo) [reply src/Plivo/Resources/Message/MessageInterface.php:135]: is there respective unit case changes also done?

File: src/Plivo/Resources/Message/MessageInterface.php [hunk 3]
```diff
@@ -144,66 +132,101 @@ protected function getList($optionalArgs = [])
...
+            if (!array_key_exists('src', $arguments) && array_key_exists('powerpackUUID', $arguments))
+            {
+                throw new PlivoValidationException("Both powerpack_uuid and src cannot be specified. Specify either powerpack_uuid or src in request params to send a message.");
+            }
 
-        if (is_null($src) &&  is_null($powerpackUUID)) {
-            throw new PlivoValidationException(
-                "Specify either powerpack_uuid or src in request params to send a message."
-            );
+            foreach ($dest as $dst)
```
Comment (narayana-plivo) [inline src/Plivo/Resources/Message/MessageInterface.php:151]: if the destination param is array, can we keep same across all the sdk?

File: src/Plivo/Resources/Message/MessageInterface.php [hunk 4]
```diff
@@ -144,66 +132,101 @@ protected function getList($optionalArgs = [])
...
         }
 
-        if (!is_null($src) && !is_null($powerpackUUID)) {
-            throw new PlivoValidationException(
-                "Both powerpack_uuid and src cannot be specified. Specify either powerpack_uuid or src in request params to send a message."
-            );
-        }
+        else
+        {
+            $src = $arguments[0];
```
Comment (narayana-plivo) [inline src/Plivo/Resources/Message/MessageInterface.php:183]: is there anywhere you mentioned the order for argument?

Comment (huzaif-plivo) [reply src/Plivo/Resources/Message/MessageInterface.php:175]: No, this wasn't mentioned earlier as well. Hence coming up with the new interface. This snippet is to handle the old interface

File: src/Plivo/Resources/Message/MessageInterface.php (hunk 3 above)
Comment (huzaif-plivo) [reply src/Plivo/Resources/Message/MessageInterface.php]: Fixed.

File: src/Plivo/Resources/Message/MessageInterface.php (hunk 2 above)
Comment (huzaif-plivo) [reply src/Plivo/Resources/Message/MessageInterface.php:135]: __call() is used here for method overloading. No changes required to update the test case.

File: src/Plivo/Resources/Message/MessageInterface.php (hunk 1 above)
Comment (huzaif-plivo) [reply src/Plivo/Resources/Message/MessageInterface.php]: Fixed

File: src/Plivo/Resources/Message/MessageInterface.php [hunk 5]
```diff
@@ -144,66 +132,93 @@ protected function getList($optionalArgs = [])
...
+            if (array_key_exists('src', $arguments) && array_key_exists('powerpack_uuid', $arguments))
+            {
+                throw new PlivoValidationException("Both powerpack_uuid and src cannot be specified. Specify either powerpack_uuid or src in request params to send a message.");
+            }
 
-        if (is_null($src) &&  is_null($powerpackUUID)) {
-            throw new PlivoValidationException(
-                "Specify either powerpack_uuid or src in request params to send a message."
-            );
+            $response = $this
```
Comment (narayana-plivo) [inline src/Plivo/Resources/Message/MessageInterface.php:150]: just move out of the if condition. same code will be used in else block also. It makes clean and less code.

Comment (huzaif-plivo) [reply src/Plivo/Resources/Message/MessageInterface.php]: updated

**Review by narayana-plivo — APPROVED** [review]

**Pull request merged**